src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -115,7 +207,43 @@ static ucs_status_t uct_rdamcm_cm_ep_client_init(uct_rdmacm_cm_ep_t *cep,
 static ucs_status_t uct_rdamcm_cm_ep_server_init(uct_rdmacm_cm_ep_t *cep,
                                                  const uct_ep_params_t *params)
 {
-    return UCS_ERR_NOT_IMPLEMENTED;
+    struct rdma_cm_event *event = (struct rdma_cm_event *)params->conn_request;
+    struct rdma_conn_param conn_param;
+    uct_rdmacm_priv_data_hdr_t hdr;
+    ucs_status_t status;
+
+    cep->wireup.server.connect_cb = params->sockaddr_connect_cb.server;
+
+    /* TODO: migrate id if cm is different */
+    ucs_assert(event->listen_id->channel == uct_rdmacm_cm_ep_get_cm(cep)->ev_ch);
+
+    cep->id          = event->id;
+    cep->id->context = cep;
+
+    status = uct_rdmacm_cm_ep_conn_param_init(cep, &conn_param);
+    if (status != UCS_OK) {
+        hdr.status = status;
+        hdr.length = 0;
+        if (rdma_reject(event->id, &hdr, sizeof(hdr))) {
```

Comment:
add wrapper for rdma_reject which also defines hdr to pass
will remove some code duplication